Mage/src/main/java/mage/cards/CardImpl.java

```diff
@@ -747,6 +753,9 @@ public boolean addCounters(Counter counter, UUID playerAddingCounters, Ability s
                 GameEvent addedAllEvent = GameEvent.getEvent(GameEvent.EventType.COUNTERS_ADDED, objectId, source, playerAddingCounters, counter.getName(), amount);
                 addedAllEvent.setFlag(isEffectFlag);
                 game.fireEvent(addedAllEvent);
+                if (informPlayers && !game.isSimulation()) {
+                    game.informPlayers(CardUtil.getAddRemoveCountersLogMessage(game, source, playerAddingCounters, finalAmount, counter.getName(), this.getLogName(), false));
```

Comment:
I think it would be worth refactoring this further so that the `.informPlayers` call happens inside `getAddRemoveCountersLogMessage`

Reply:
Done.